### PR TITLE
layers: Fix AccelerationStructureDescriptor::is_khr_ initialisation

### PR DESCRIPTION
**Pull request by arno-lunarg**
- Fix for https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/5300

AccelerationStructureDescriptor::is_khr_ was not initialised

Locally, I now have no validation error in this test, but it still fails:
```
Writing test log into TestResults.qpa
dEQP Core vulkan-cts-1.3.5.0-238-ge873c6862 (0xe873c686) starting..
  target implementation = 'Default'
loader_get_json: Failed to open JSON file C:\Program Files (x86)\Epic Games\Epic Online Services\managedArtifacts\98bc04bc842e4906993fd6d6644ffb8d\EOSOverlayVkLayer-Win64.json

Test case 'dEQP-VK.binding_model.mutable_descriptor.single_nonmutable.acceleration_structure_khr.update_copy.mutable_source.normal_source.pool_same_types.pre_update.no_array.comp'..
  InternalError (1 API usage errors found)

DONE!

Test run totals:
  Passed:        0/1 (0.0%)
  Failed:        1/1 (100.0%)
  Not supported: 0/1 (0.0%)
  Warnings:      0/1 (0.0%)
  Waived:        0/1 (0.0%)
```

I run with the following command line options:
--deqp-validation=enable --deqp-print-validation-errors --deqp-caselist=dEQP-VK.binding_model.mutable_descriptor.single_nonmutable.acceleration_structure_khr.update_copy.mutable_source.normal_source.pool_same_types.pre_update.no_array.comp

### CODE REVIEW

**Review by artem-lunarg — APPROVED** [review]
LGTM!

**Comment by artem-lunarg** [comment]
One idea is to get rid of `is_khr_` state and to use only getters, like the one introduced here.

**Comment by arno-lunarg** [comment]
> One idea is to get rid of `is_khr_` state and to use only getters, like the one introduced here.

Yes I like that, I am going to try to do this

**Comment by arno-lunarg** [comment]
I am also in the process of adding a positive test

File: layers/state_tracker/descriptor_sets.h [hunk 1]
```diff
@@ -656,12 +656,17 @@ class MutableDescriptor : public Descriptor {
...
-    VkAccelerationStructureKHR GetAccelerationStructure() const { return acc_; }
+    VkAccelerationStructureKHR GetAccelerationStructureKHR() const { return acc_; }
     const ACCELERATION_STRUCTURE_STATE_KHR *GetAccelerationStructureStateKHR() const { return acc_state_.get(); }
     ACCELERATION_STRUCTURE_STATE_KHR *GetAccelerationStructureStateKHR() { return acc_state_.get(); }
     VkAccelerationStructureNV GetAccelerationStructureNV() const { return acc_nv_; }
     const ACCELERATION_STRUCTURE_STATE *GetAccelerationStructureStateNV() const { return acc_state_nv_.get(); }
     ACCELERATION_STRUCTURE_STATE *GetAccelerationStructureStateNV() { return acc_state_nv_.get(); }
+    bool IsAccelerationStructureKHR() const {
+        auto acc_khr = GetAccelerationStructureKHR();
+        assert(acc_khr != VK_NULL_HANDLE || GetAccelerationStructureNV() != VK_NULL_HANDLE);
```
Comment (spencer-lunarg) [inline layers/state_tracker/descriptor_sets.h:667]: It is not obvious why `GetAccelerationStructureNV()` needs to be checked her in this `KHR` call, might be worth adding a comment

**Review by spencer-lunarg — APPROVED** [review]

File: layers/state_tracker/descriptor_sets.h (hunk 1 above)
Comment (artem-lunarg) [reply layers/state_tracker/descriptor_sets.h:667]: +1 
I started to write the same comment earlier today but then changed my mind. But it took few minutes to understand semantics, so it's a good idea to add it.

**Comment by arno-lunarg** [comment]
> > One idea is to get rid of `is_khr_` state and to use only getters, like the one introduced here.
> 
> Yes I like that, I am going to try to do this

Following the problem I raised in the team chat, I am not going to touch the `is_khr_` since it may be invalid code

**Pull request merged**